airflow-core/docs/security/audit_logs.rst

```diff
@@ -18,50 +18,607 @@
+- **Performance Monitoring**: Recording execution times, resource usage, and system metrics
+- **Operational Insights**: Tracking system health, component interactions, and workflow execution
+- **Development Support**: Offering detailed information for code debugging and optimization
+
+Event logs are typically stored in log files or external logging systems and include information such as:
+
+- Task execution details and output
+- System errors and warnings
+- Performance metrics and timing information
+- Component startup and shutdown events
+- Resource utilization data
+
+Audit Logs vs Event Logs
+------------------------------------------
+
+While both logging systems are crucial for system management, they serve distinct purposes and audiences:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 37 38
+
+   * - Characteristic
+     - Audit Logs
+     - Event Logs
+   * - **Primary Purpose**
+     - Accountability and compliance tracking
+     - Operational monitoring and system debugging
+   * - **Target Audience**
+     - Security teams, auditors, compliance officers
+     - Developers, system administrators, operations teams
+   * - **Content Focus**
+     - User actions and administrative changes
+     - System behavior, errors, and performance data
+   * - **Storage Location**
+     - Structured database table (``log``)
+     - Log files, external logging systems
+   * - **Retention Requirements**
+     - Long-term (months to years for compliance)
+     - Short to medium-term (days to weeks)
+   * - **Query Patterns**
+     - "Who modified this configuration?"
+     - "Why did this task execution fail?"
+
+
+Accessing Audit Logs
 --------------------
 
-Audit logs exist at the task level and the user level.
+Airflow provides multiple interfaces for accessing audit log data, each suited to different use cases and technical requirements:
 
-- Task Level: At the task level, audit logs capture information related to the execution of a task, such as the start time, end time, and status of the task.
+**Web User Interface**
+   The Airflow web interface provides the most accessible method for viewing audit logs. Navigate to **Browse → Audit Logs** to access an interface with built-in filtering, sorting, and search capabilities. This interface is ideal for ad-hoc investigations and routine monitoring.
 
-- User Level: At the user level, audit logs capture information related to user actions, such as creating, modifying, or deleting a Dag or task.
+**REST API Integration**
+   For programmatic access and system integration, use the ``/eventLogs`` REST API endpoint. This approach enables automated monitoring, integration with external security tools, and custom reporting applications.
 
+**Direct Database Access**
+   Advanced users can query the ``log`` table directly using SQL. This method provides maximum flexibility for complex queries, custom reporting, and integration with business intelligence tools.
```

Comment:
We should not recommend quering the database directly. This is not a public API. Everybody can make this at own risk only.